c/tests/test_tables.c

```diff
@@ -324,6 +324,143 @@ test_table_collection_simplify_errors(void)
+    ret = tsk_table_collection_init(&tc1, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+
+    tc1.reference_sequence = tsk_malloc(sizeof(tsk_reference_sequence_t));
+    CU_ASSERT_NOT_EQUAL_FATAL(tc1.reference_sequence, NULL);
+    tsk_memset(tc1.reference_sequence, 0, sizeof(tsk_reference_sequence_t));
+
+    ret = tsk_reference_sequence_set_data(
+        tc1.reference_sequence, example_data, example_data_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_copy(&tc1, &tc2, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_TRUE(tsk_table_collection_equals(&tc1, &tc2, 0));
+
+    ret = tsk_reference_sequence_set_url(
+        tc1.reference_sequence, example_url, example_url_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_copy(&tc1, &tc2, TSK_NO_INIT);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_TRUE(tsk_table_collection_equals(&tc1, &tc2, 0));
+
+    ret = tsk_reference_sequence_set_metadata(
+        tc1.reference_sequence, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_copy(&tc1, &tc2, TSK_NO_INIT);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_TRUE(tsk_table_collection_equals(&tc1, &tc2, 0));
+
+    ret = tsk_reference_sequence_set_metadata_schema(
+        tc1.reference_sequence, example_schema, example_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_copy(&tc1, &tc2, TSK_NO_INIT);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_TRUE(tsk_table_collection_equals(&tc1, &tc2, 0));
+
+    // Test dump and load
+    tsk_table_collection_free(&tc1);
+    tsk_table_collection_free(&tc2);
+    ret = tsk_table_collection_init(&tc1, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    tc1.sequence_length = 1.0;
+
+    tc1.reference_sequence = tsk_malloc(sizeof(tsk_reference_sequence_t));
+    CU_ASSERT_NOT_EQUAL_FATAL(tc1.reference_sequence, NULL);
+    tsk_memset(tc1.reference_sequence, 0, sizeof(tsk_reference_sequence_t));
+
+    ret = tsk_reference_sequence_set_data(
+        tc1.reference_sequence, example_data, example_data_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_reference_sequence_set_url(
+        tc1.reference_sequence, example_url, example_url_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_reference_sequence_set_metadata(
+        tc1.reference_sequence, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_reference_sequence_set_metadata_schema(
+        tc1.reference_sequence, example_schema, example_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_dump(&tc1, _tmp_file_name, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_load(&tc2, _tmp_file_name, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_TRUE(tsk_table_collection_equals(&tc1, &tc2, 0));
+    tsk_table_collection_free(&tc1);
+    tsk_table_collection_free(&tc2);
```

Comment:
Looks good, but not clear to me what bits are testing the table_collection and what bits are testing ``reference_sequence``. Perhaps better to abstract out the operations we need to ``tsk_reference_sequence_t`` and test that outside of the context of the table collection, *then* test the table collection operations that depend on the reference sequence?

Reply:
Done